src/Routes.jsx

```diff
@@ -39,27 +69,48 @@ class Routes extends Component {
       data,
       iconAtlas: atlasIcon,
       iconMapping: ICON_MAPPING,
+      pickable: true,
+      // calls pop-up function
+      onClick: info => this.displayVehicleInfo(info),
     }));
   }
 
+  displayVehicleInfo(info) {
+    /* calls parent' onMarkerClick function to show pop-up to display vehicle id & heading info */
+
+    if (info && info.object && info.object.vid) {
+      this.props.onMarkerClick(
+        info.lngLat[0],
```

Comment:
what's lngLat[0]? is it the lng of info?

Reply:
yep, it's DeckGL format.

https://uber.github.io/deck.gl/#/documentation/getting-started/adding-interactivity?section=the-picking-info-object